src/pages/index.js

```diff
@@ -1,67 +1,263 @@
+// -------------------------------------------------------------------------------------
+api
+  .getCardList()
+  .then((cardList) => {
+    cardSection.renderItems(cardList);
+  })
+  .catch((error) => {
+    console.error("Failed to fetch or render cards:", error);
+  });
+
+// -------------------------------------------------------------------------------------
+// Additional Card adding to API & DOM
+// -------------------------------------------------------------------------------------
 const addCardPopup = new PopupWithForm("#add-card-modal", {
   handleFormSubmit: (formData) => {
     const cardData = {
       name: formData["title"],
       link: formData["Image URL"],
     };
-    const card = createCard(cardData);
-    cardSection.prependItem(card);
-    addCardPopup.getForm().reset();
-    formValidators[
-      addCardPopup.getForm().getAttribute("name")
-    ].toggleButtonState();
-    addCardPopup.close();
+
+    addCardPopup.toggleUploadIndicator(true);
+
+    api
+      .addCard(cardData)
+      .then((apiCardData) => {
+        const card = new Card(
+          apiCardData,
+          cardTemplate,
+          (data) => imagePopup.open(data),
+          handleTrashClick,
+          updateLike
+        ).getView();
+
+        cardSection.prependItem(card);
+        addCardPopup.close();
+      })
+      .catch((error) => console.error("Error adding card:", error))
+      .finally(() => addCardPopup.toggleUploadIndicator(false));
   },
 });
+
 addCardPopup.setEventListeners();
 
-const imagePopup = new PopupWithImage("#image-modal");
-imagePopup.setEventListeners();
 // -------------------------------------------------------------------------------------
-//                               Validation
+// Card Liking
+// -------------------------------------------------------------------------------------
+
+function updateLike(card) {
+  const isCurrentlyLiked = card._likeButton.classList.contains(
+    "card__like-button_active"
+  );
+
+  if (isCurrentlyLiked) {
+    api
+      .unlikeCard(card._id)
+      .then(() => {
+        card.toggleIsLiked(false);
+      })
+      .catch((error) => {
+        console.error("Error unliking card:", error);
+      });
+  } else {
+    api
+      .likeCard(card._id)
+      .then(() => {
+        card.toggleIsLiked(true);
+      })
+      .catch((error) => {
+        console.error("Error liking card:", error);
+      });
+  }
+}
+
+// -------------------------------------------------------------------------------------
+// Card deletion from API & DOM
+// -------------------------------------------------------------------------------------
+const deleteCardModal = new PopupWithConfirm("#delete-card-modal", () => {});
+deleteCardModal.setEventListeners();
+
+function handleTrashClick(card) {
+  deleteCardModal.open();
+  deleteCardModal.setSubmitHandler(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        card._element.remove();
+        card._element = null;
```

Comment:
`_element`

* Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.